authentication/native/server.go

```diff
@@ -0,0 +1,117 @@
+package native
+
+import (
+	"context"
+	"time"
+
+	goCore "github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	crypto "github.com/ElrondNetwork/elrond-go-crypto"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/authentication"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/workflows"
+)
+
+// ArgsNativeAuthServer is the DTO used in the native auth server constructor
+type ArgsNativeAuthServer struct {
+	Proxy           workflows.ProxyHandler
+	TokenHandler    authentication.AuthTokenHandler
+	Signer          crypto.SingleSigner
+	PubKeyConverter goCore.PubkeyConverter
+	KeyGenerator    crypto.KeyGenerator
+	AcceptedHosts   map[string]struct{}
+}
+
+type authServer struct {
+	proxy           workflows.ProxyHandler
+	tokenHandler    authentication.AuthTokenHandler
+	signer          crypto.SingleSigner
+	keyGenerator    crypto.KeyGenerator
+	pubKeyConverter goCore.PubkeyConverter
+	acceptedHosts   map[string]struct{}
+	getTimeHandler  func() time.Time
+}
+
+// NewNativeAuthServer returns a native authentication server
```

Comment:
Maybe we should expand this comment a bit - explaining what the auth server actually can do and how

Reply:
done